src/ntfsck.c

```diff
@@ -3144,14 +3183,39 @@ static void ntfsck_validate_index_blocks(ntfs_volume *vol,
 
 			for (; (u8 *)ie < index_end;
 				ie = (INDEX_ENTRY *)((u8 *)ie + le16_to_cpu(ie->length))) {
-				/* Bounds checks. */
-				if (((u8 *)ie < (u8 *)ia) ||
+
+				/* check bitmap for sub-node */
+				if (ie->ie_flags & INDEX_ENTRY_NODE) {
+					VCN vcn = ntfs_ie_get_vcn(ie);
+					u32 sub_bmp_pos;
+
+					sub_bmp_pos = (vcn << ictx->vcn_size_bits) / ictx->block_size;
```

Comment:
we can move it to above if (max_vcn_bits <= vcn).

Reply:
I moved it above checking sub-node